### PR TITLE
Have the scheduled_id/at columns be specified in the schema, not by the column name

### LINKED ISSUE #1816: Schedule schema should find scheduled_at/id by schema field, not by name
P1(?)

right now, we know which field is `scheduled_at` and `scheduled_id` because we find them by that specific name - instead, there should be a `scheduled_at_col: ColId` field in the schema, and scheduled_id should just be the primary_key.

### PR DESCRIPTION
**Pull request by coolreader18**
# Description of Changes

Resolves #1816. Also changes `primary_key` to be a `ColList` for future compat reasons.

# API and ABI breaking changes

Breaking, yes, but so we can be compatible with future changes.

### CODE REVIEW

File: crates/core/src/host/scheduler.rs [hunk 1]
```diff
@@ -36,8 +34,17 @@ pub struct ScheduledReducerId {
     table_id: TableId,
     /// The particular schedule row in the reducer scheduling table referred to by `self.table_id`.
     schedule_id: u64,
+    // These may seem redundant, but they're actually free - they fit in the struct padding.
```
[the remark this answers is not in the document]
Comment (coolreader18) [reply crates/core/src/host/scheduler.rs:37]: This is never serialized to BSATN, though.

File: crates/schema/src/def.rs [hunk 2]
```diff
@@ -467,7 +467,7 @@ impl From<TableDef> for RawTableDefV9 {
         RawTableDefV9 {
             name: name.into(),
             product_type_ref,
-            primary_key,
+            primary_key: ColList::from_iter(primary_key),
```
[the remark this answers is not in the document]
Comment (coolreader18) [reply crates/schema/src/def.rs:470]: `primary_key` is an `Option` here; I don't think this will compile.

File: crates/bindings-macro/src/lib.rs [hunk 3]
```diff
@@ -1029,7 +1029,19 @@ fn table_impl(mut args: TableArgs, mut item: MutItem<syn::DeriveInput>) -> syn::
     let unique_col_ids = unique_columns.iter().map(|col| col.index);
     let primary_col_id = primary_key_column.iter().map(|col| col.index);
     let sequence_col_ids = sequenced_columns.iter().map(|col| col.index);
-    let scheduled_reducer_ident = args.scheduled.iter();
+
+    let schedule = args
+        .scheduled
+        .as_ref()
+        .map(|reducer| {
+            // scheduled_at was inserted as the last field
```
Comment (kazimuth) [inline crates/bindings-macro/src/lib.rs:1037]: I was wondering if we wanted to have the users declare the column as well so they can see what is going on + sometimes get better intellisense, but that could be done as a follow-up PR if we decide to do it.

**Review by kazimuth — APPROVED** [review]
Happy with this, the parts of it touching the stuff I'm familiar with all look good.

File: crates/bindings-macro/src/lib.rs (hunk 3 above)
Comment (coolreader18) [reply crates/bindings-macro/src/lib.rs:1037]: Yes that's the plan!

**Pull request merged**